File: frontend/projects/irrtirer/src/app/shared/mosaic-generator/sectors/intersections/intersections.ts
import { IntersectionRecord } from "./intersection-record";

export class Intersections {
    records: IntersectionRecord[] = [];
    intersectionArea: number = 0;

    public addIntersection(record: IntersectionRecord): void {
        this.records.push(record);
        this.intersectionArea += record.area;
    }

    public static restore(obj: Intersections): void {
        Object.setPrototypeOf(obj, Intersections.prototype);
    }
}

File: backend/Irrtirer/Irrtirer.Generator/MosaicStructure/Intersections/Intersections.cs
using System.Collections.Generic;

namespace Irrtirer.Generator.MosaicStructure
{
    internal class Intersections
    {
        public List<IntersectionRecord> Records { get; private set; } = new List<IntersectionRecord>();
        public float IntersectionArea { get; private set; }

        public void AddIntersection(IntersectionRecord record)
        {
            Records.Add(record);
            IntersectionArea += record.Area;
        }
    }
}
